#940 upgrade papercut and changed portnumbers

anoordover opened the pull request:
<!-- 🚨 Please Do Not skip any instructions and information mentioned below as they are all required and essential to evaluate and test the PR. By fulfilling all the required information you will be able to reduce the volume of questions and most likely help merge the PR faster 🚨 -->

<!-- ⚠️ We will not merge the PR to the main repo if your changes are not in a *feature branch* of your forked repository. Create a new branch in your repo, **do not use the `main` branch in your repo**! -->

<!-- ⚠️ **Every PR** needs to have a linked issue and have previously been approved. PRs that don't follow this will be rejected. >

<!-- 👉 It is imperative to resolve ONE ISSUE PER PR and avoid making multiple changes unless the changes interrelate with each other -->

<!-- 📝 Please always keep the "☑️ Allow edits by maintainers" button checked in the Pull Request Template as it increases collaboration with the Toolkit maintainers by permitting commits to your PR branch (only) created from your fork. This can let us quickly make fixes for minor typos or forgotten StyleCop issues during review without needing to wait on you doing extra work. Let us help you help us! 🎉 -->

**Closes #940**

<!-- Add an overview of the changes here -->

<!-- All details should be in the linked issue. Feel free to call out any outstanding differences here. -->

## PR Checklist

<!-- Please check if your PR fulfills the following requirements, and remove the ones that are not applicable to the current PR -->

- [x] Created a feature/dev branch in your fork (vs. submitting directly from a commit on main)
- [x] Based off latest main branch of toolkit
- [x] PR doesn't include merge commits (always rebase on top of our main, if needed)
- [ ] New integration
  - [ ] Docs are written
  - [ ] Added description of major feature to project description for NuGet package (4000 total character limit, so don't push entire description over that)
- [x] Tests for the changes have been added (for bug fixes / features) (if applicable)
- [x] Contains **NO** breaking changes
- [x] Every new API (including internal ones) has full XML docs
- [x] Code follows all style conventions

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below.
Please note that breaking changes are likely to be rejected within minor release cycles or held until major versions. -->

## Other information

@mguinness thanks for the pointer to the changed ports.
I don't know if this is a breaking change for this component.

<!-- Please add any other information that might be helpful to reviewers. -->

mguinness commented:
LGTM, I did some testing and revising the ports worked as expected.

The only problem I found was that I had to replace `httpPort` with `(int)HttpStatusCode.OK` in the following:

https://github.com/CommunityToolkit/Aspire/blob/4fc8f807ef9ea9107ce92f9b917fb4d72b75d322/src/CommunityToolkit.Aspire.Hosting.PapercutSmtp/PapercutSmtpHostingExtension.cs#L38

Also do you think it's worth adding some example code for the connection string to README?

```c#
var connStr = config.GetConnectionString("papercut");
var uri = new Uri(connStr[9..]);
var sc = new SmtpClient(uri.Host, uri.Port);
```

anoordover commented:
@mguinness thanks for pointing out this bug which might have been present from the start of this component.
I remove the status parameter from the call because the default value is 200.
@mguinness how did you notice this bug? Can I add a unittest to test this?
Do you have any pointers for me?

mguinness commented:
I noticed the bug when I changed AddPapercutSmtp to specify ports.  

`var papercut = builder.AddPapercutSmtp("papercut", 8080, 2525)`

Then health check then failed, but I'm not sure why it didn't occur when `httpPort` variable was 80.

The message in [UriHealthCheck](
https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/blob/846fbc99bcf60eafea23096c9aab05b8c8459b6b/src/HealthChecks.Uris/UriHealthCheck.cs#L56-L59) should be returned when status code 80 is incorrectly expected.  @sungam3r Any ideas?

mguinness commented:
> What do you mean?

Apologies, I wasn't very clear.  I suppose the question should be whether bounds checking occur in `ExpectedHttpCodes` to ensure valid status code >= 100 and <=599 in [UriHealthCheckOptions](
https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/blob/846fbc99bcf60eafea23096c9aab05b8c8459b6b/src/HealthChecks.Uris/UriHealthCheckOptions.cs#L91) before `httpClient.SendAsync` is called?

anoordover commented:
@mguinness I have a local change ready to commit that I can push if we want to test whether setting the portnumbers on the papercut host will effect the healthcheck.

```csharp
var papercut = builder.AddPapercutSmtp(PapercutConstants.ResourceNameDefaultParameters);
var papercut2 = builder.AddPapercutSmtp(PapercutConstants.ResourceNameOverrideParameters, httpPort: 8080, smtpPort: 2525);
```

Now I can use the two resource-names in my unittest in a Theory. My previous code would fail on the second papercut-resource.
Shall I commit this?

Changes will be:
- Added a static class PapercutConstants containing the names of the two papercut resources;
- Added a second papercut resource in my example project;
- Changed my AppHostTests to run the test ResourceStartsAndMailShouldBeReceived as a Theory with the two resourcenames.

mguinness commented:
@aaronpowell might be better placed to answer.  I suppose you could test that a port number outside the range 1-65535 fails.

aaronpowell commented:
not sure I'm properly following - is the issue that if you set a port number outside a certain range things break unexpectedly?

mguinness commented:
I just tested and [AllocatedEndpoint](https://github.com/dotnet/aspire/blob/a79afe3c1d0bc7d60b1905a36a2b27d4c2815080/src/Aspire.Hosting/ApplicationModel/AllocatedEndpoint.cs#L58-L59) throws a `ArgumentOutOfRangeException`.

Arno, I think what you already have in the PR is sufficient for merging if it can be assigned for review.

aaronpowell commented:
Port range management should delegate to Aspire, unless a resource has very specific requirements
